lib/Minion/Backend/Pg.pm

```diff
@@ -178,9 +178,18 @@ sub repair {
   my $minion = $self->minion;
   $db->query("DELETE FROM minion_workers WHERE notified < NOW() - INTERVAL '1 second' * ?", $minion->missing_after);
 
-  # Old jobs
-  $db->query("DELETE FROM minion_jobs WHERE state = 'finished' AND finished <= NOW() - INTERVAL '1 second' * ?",
-    $minion->remove_after);
+  # Old jobs with no unresolved dependencies and expired jobs
+  $db->query(
+    "DELETE FROM minion_jobs
+     WHERE id IN (
+      (
+        SELECT id FROM minion_jobs WHERE state = 'finished' AND finished <= NOW() - INTERVAL '1 second' * ?
+        EXCEPT
+        SELECT unnest(parents) AS id FROM minion_jobs WHERE state != 'finished'
+      )
+      UNION
+      SELECT id FROM minion_jobs WHERE state = 'inactive' AND expires <= NOW()
```

Comment:
Looks like you're deleting expired jobs twice now.

Reply:
an oversight, fixed in recent commit (including failed perltidy check)

Reply:
Are you sure combining the two queries is more efficient here?

Reply:
For debugging alone two queries would seem much better.

Reply:
> Are you sure combining the two queries is more efficient here?

I would think its always more efficient to combine it into a single query. But I would have to have a chat with my dba.

> For debugging alone two queries would seem much better.

I agree here it would be better for debugging

The original cleanup of expired jobs, is fast by it self

Reply:
I'm also concerned about the time locks are being kept on the table. Even if the large transaction is theoretically a tiny bit more efficient, multiple smaller deletes would be preferable if they allow for faster updates from other transactions.

Reply:
this is reflected in the push from earlier today